nats.go

```diff
@@ -1682,9 +1682,11 @@ slowConsumer:
 // permissions violation on either publish or subscribe.
 func (nc *Conn) processPermissionsViolation(err string) {
 	nc.mu.Lock()
-	nc.err = errors.New("nats: " + err)
+	// create error here so we can pass it as a closure to the async cb dispatcher.
+	e := errors.New("nats: " + err)
```

Comment:
Unrelated to this, but IMO we should consider using "typed" errors. Users should not have to resort to parsing error strings to handle different errors.